cpp/src/arrow/compute/kernels/aggregate_basic.inc

```diff
@@ -0,0 +1,1025 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
Can we name this file `.inc.cc` or something? The `.inc` suffix may not be enough to automatically get proper type detection in text editors.